### ISSUE TITLE
Inappropriate sampling strategy in validation epoch may cause unreliable dices

### ISSUE THREAD

**Issue by on1262** [MIC-DKFZ/nnUNet#1729]
Thanks for your great work of nnunet! I can feel your enthusiastics from some comments :)

However, when I applying nnunetv2 to my image segmentation project, I found the validation dice is always higher than the dice calculated from output images. Usually I get 5%-8% higher dice. Finally I found there are some inappropriate sampling strategies in the configuration of validation dataloader.

```
dl_val = nnUNetDataLoader2D(dataset_val, self.batch_size,
                                         self.configuration_manager.patch_size,
                                         self.configuration_manager.patch_size,
                                         self.label_manager,
                                         oversample_foreground_percent=self.oversample_foreground_percent,
                                         sampling_probabilities=None, pad_sides=None)
```

The validation dataloader in `nnUNetTrainer.py` is configured as above. 

First, applying oversample foreground will cause statistics bias. In default it is set to 0.33, which will let the model just watching a small area with foreground labels in 33% of overall samples. 

Secondly, images in validation epoch are splited into patches. These patches are randomly chosen. I still observed a small bias (about 3%) when I set oversample coeff to zero. I do not know why. Maybe not all pixels have the same possibility to be chosen. 

At last, by default we only choose 50 patches for validation. This number could be too small and estimated dice could be inaccurate. Additionally, validation batchsize can be larger. Training batchsize is set near 12 to prevent overfitting. But it is unnecessary in validation and we can use more VRAM.

**Comment by saikat-roy** [comment]
Hi @on1262 . Thank you for the comments. The validation process is designed to observe change on a validation set but not to actually base decisions on it. As you have rightly pointed out, the patches are selected randomly. This is not really designed to cover the entire validation set - thus this _noisy_ validation is only supposed to be used for a lose estimation of learning.

You are of course free to change this by defining a new trainer if you want if you want to have better visualization of the loss curves or want the estimated dice to be closer to your final validation dice, but just to be clear, the final checkpoint is not saved based on this estimated dice. So overall training behavior should not change.
